components/gitpod-protocol/src/public-api-converter.spec.ts

```diff
@@ -91,6 +91,37 @@ describe("PublicAPIConverter", () => {
         });
     });
 
+    describe("fromPartialConfiguration", () => {
+        it("should convert a configuration name change to a Project", () => {
+            const config: PartialConfiguration = {
+                id: "123",
+                name: "My Config",
+            };
+            const result = converter.fromPartialConfiguration(config);
+            expect(result.id).to.equal(config.id);
+            expect(result.settings).to.be.undefined;
+        });
+        it("should carry over Configuration settings correctly", () => {
+            const config: PartialConfiguration = {
+                id: "123",
+                name: undefined,
+                workspaceSettings: {
+                    workspaceClass: "huge",
+                },
+                prebuildSettings: {
+                    enabled: true,
+                    prebuildInterval: 5,
+                },
+            };
+            const result = converter.fromPartialConfiguration(config);
+            expect(result.id).to.equal(config.id);
+            expect(result.settings?.workspaceClasses?.regular).to.deep.equal("huge");
```

Comment:
should not we test that other properties are not present?

Reply:
https://github.com/gitpod-io/gitpod/pull/19080/commits/c21154c5a9db6dc1f05ce44286b4169a793aefec